nbconverted/3_park_analysis.py

```diff
@@ -391,3 +413,277 @@ def get_counts(status):
 # 
 # The next step is to break this down by gene - are there genes/cancer types where the functional effect of the "two hits" is clearer, or less clear? Are there genes where we see "two hits" at the genetic level but no functional effect/classifier difference, or are the sources of information basically redundant?
+
+# ### Statistical testing for individual gene/cancer type combinations
+
+# In[26]:
+
+
+info_compare_df = ut.test_all(park_info_df)
+print(info_compare_df.shape)
+print('reject null for:', info_compare_df.reject_null.sum(), '/', info_compare_df.shape[0])
+info_compare_df.sort_values(by='corr_pval', ascending=True).head()
+
+
+# In[27]:
+
+
+# plot top three most significant
+# and top three least significant
+order = ['none', 'one', 'both']
+def get_counts(status):
+    un = np.unique(status, return_counts=True)
+    return {s: c for s, c in zip(*un)}
+
+sns.set({'figure.figsize': (24, 16)})
+fig, axarr = plt.subplots(2, 3)
+
+valid_df = info_compare_df[info_compare_df.p_value != 1.0]
+plot_ids = (
+    valid_df.sort_values(by='corr_pval', ascending=True).identifier.tolist()[:3] + 
+    valid_df.sort_values(by='corr_pval', ascending=False).identifier.tolist()[:3]
+)
+
+for ix, identifier in enumerate(plot_ids):
+    ax = axarr[ix // 3, ix % 3]
+    plot_df = park_info_df[park_info_df.identifier == identifier]
+    sns.boxplot(data=plot_df, x='status', y='positive_prob', order=order, ax=ax)
+    count_map = get_counts(plot_df.status.values)
+    ax.set_xticks(np.arange(3),
+               ['{} (n={})'.format(l, count_map[l]) for l in order])
+    ax.set_title(identifier)
+
+
+# ### Compare classifier-based statistical testing vs. Park et al statistical testing
+
+# In[28]:
+
+
+pair_df = (info_compare_df
+    .merge(park_df, left_on='identifier', right_index=True)
+    .rename(columns={'corr_pval': 'classifier_pval',
+                     'FDR': 'park_pval'})
+)
+print(pair_df.shape)
+print(pair_df['class'].unique())
+print(pair_df.classifier_pval.isna().sum())
+pair_df.head()
+
+
+# In[29]:
+
+
+class_order = ['class 1', 'class 2', 'class 3', 'class 4']
+sns.set({'figure.figsize': (8, 6)})
+sns.scatterplot(data=pair_df, x='classifier_pval', y='park_pval',
+                hue='class', hue_order=class_order)
+plt.xlim(-0.1, 1.1)
+plt.ylim(-0.1, 1.1)
+plt.xlabel('Classifier p-value')
+plt.ylabel('Park et al. p-value')
+plt.title('Classifier vs. Park p-value, all Park genes')
+
+
+# In[30]:
+
+
+sns.set({'figure.figsize': (8, 6)})
+sns.scatterplot(data=pair_df, x='classifier_pval', y='park_pval',
+                hue='class', hue_order=class_order)
+plt.xscale('log')
+plt.yscale('log')
+plt.xlim(10**-10, 10**0+1)
+plt.ylim(10**-10, 10**0+1)
+plt.xlabel('Classifier p-value')
+plt.ylabel('Park et al. p-value')
+plt.title('Classifier vs. Park p-value, all Park genes')
```

Comment:
Now I'm just being persnickety, but I'm weirded out by the direction of this plot. I'm used to up and right = greater significance. If you're going to use this plot in a paper, any chance you could flip the axis directions?

Reply:
Yeah, I'll just flip them now actually. That makes more sense to me too.